src/uct/ib/dc/dc_mlx5_devx.c

```diff
@@ -50,7 +52,16 @@ ucs_status_t uct_dc_mlx5_iface_devx_create_dct(uct_dc_mlx5_iface_t *iface)
     if (!uct_ib_iface_is_roce(&iface->super.super.super)) {
         UCT_IB_MLX5DV_SET(dctc, dctc, pkey_index, ib_iface->pkey_index);
     }
-    UCT_IB_MLX5DV_SET(dctc, dctc, port, ib_iface->config.port_num);
+
+    if (iface->rx.port_affinity == 0) {
+        port_affinity = ib_iface->config.port_num;
```

Comment:
won't `ib_iface->config.port_num` always be 1 in case of RoCE LAG?
seems only possible policies are either specific number or random.
BTW, is there option to set to "hash" mode so hw will decide?

Reply:
@yosefe 
1. When running on ```hash RoCE LAG``` devices(e.g. CX6DX), the ```UCT_IB_MLX5DV_SET(dctc, dctc, port, 1 or 2)``` doesn't affect the RDMA_READ response port. ```hash RoCE LAG``` can full leverage the two slave ports for response.
2. When running on ```queue affinity RoCE LAG``` devices(e.g. CX5), the ```UCT_IB_MLX5DV_SET(dctc, dctc, port, 1 or 2)``` will affect the RDMA_READ response port.

According to the above info, what's about below design?
- ```auto``` or ```0```
    1) When running on ```queue affinity RoCE LAG``` devices(e.g. CX5), use ```random policy``` to set the port for response.
    2) When running on other RoCE LAG mode devices(e.g. ```hash RoCE LAG``` under CX6DX), keep current code without change.
- specific port number
   1) fail if devices doesn't support ```queue affinity RoCE LAG```
   2) fail if port number is outside the range (e.g. ```{1, 2}``` under CX5 RoCE LAG case)
   3) set the port number to be the specific port number.

Reply:
@yosefe 
Update design idea. What do you think?
1. This PR is useful for ```queue affinity RoCE LAG``` to give the configuration option to set DCT Context port.
2. For ```auto``` configuration, use ```random policy``` to set the DCT Context port. ```auto``` is the default configuration.
3. For ```specific port number```, fall back to ```set DCT Context port to be 1``` and output warning message if there's something wrong with the ```given port number by user```(e.g. 1) ```not under RoCE LAG case``` 2)```out of range```). If the ```given port number by user``` is right, set the ```DCT Context port``` to be the ```given port number by user```